sdk/core/azure-core-perf/src/main/java/com/azure/core/perf/BinaryDataSendTest.java

```diff
@@ -28,7 +28,7 @@ public void run() {
 
     @Override
     public Mono<Void> runAsync() {
-        return service.setBinaryData(endpoint, id, binaryDataSupplier.get(), length)
-            .then();
+        return Mono.fromSupplier(binaryDataSupplier)
+            .flatMap(data -> service.setBinaryData(endpoint, id, data, length));
```

Comment:
Just curious, why is the new code better?

Reply:
The `binaryDataSupplier` is supposed to supply fresh `BinaryData` for each `runAsync`.
The old code is effectively doing that, but only because the caller uses `flatMap` - which is just luck rather than proper reactive stream.
Wrapping the supplier in Mono makes it properly reactive, i.e. subscriber triggers binary data creation regardless of what caller of this is doing.